server/src/test/java/com/google/sps/AdRowProcessorTest.java

```diff
@@ -1,30 +1,186 @@
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad1 = AdRowProcessor.convertRowToAd(row1);
+    Assert.assertEquals(ad1.getIsTargetingAge(), false); 
+
+    String targeted = " 18-48 ";
+    String[] row2 = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k",targeted,
+                    "Not targeted","Not targeted","0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad2 = AdRowProcessor.convertRowToAd(row2);
+    Assert.assertEquals(ad2.getIsTargetingAge(), true);   
+  }
+
+  @Test
+  public void Should_AcceptTarget_When_GenderTargetValid() {
+    String validInput = " Female, Unknown gender ";
+    List<String> expected = Arrays.asList("Female","Unknown gender");
+    String[] row = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k","Not targeted",
+                    validInput,"Not targeted","0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+    Assert.assertEquals(ad.getGenderTargets(), expected); 
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void Should_ThrowException_When_GenderTargetInvalid() {
+    String invalidInput = " apples ";
+    String[] row = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k","Not targeted",
+                    invalidInput,"Not targeted","0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad = AdRowProcessor.convertRowToAd(row);
   }
 
+  @Test
+  public void Should_AcceptTarget_When_GeoTargetValid() {
+    String validInput = " California,    Alaska ";
+    List<String> expected = Arrays.asList("California","Alaska");
+    String[] row = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k","Not targeted",
+                    "Not targeted",validInput,"0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+    Assert.assertEquals(ad.getGeoTargets(), expected); 
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void Should_ThrowException_When_GeoTargetInvalid() {
+    String invalidInput = " apples ";
+    String[] row = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k","Not targeted",
+                    "Not targeted",invalidInput,"0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"};  
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+  }
```

Comment:
After the MVP demo today, should I allow for geo targets that aren't U.S. States? I didn't realize we still had zipcodes and congressional codes and stuff.